services/ux-backend/handlers/featureflags/handler.go

```diff
@@ -0,0 +1,215 @@
+package featureflags
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+	nbv1alpha1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/klog/v2"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	flagNoobaa string = "noobaa"
+	flagRGW    string = "rgw"
+)
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, client ctrlclient.Client, namespace string) {
+	switch r.Method {
+	case "GET":
+		handleGet(w, r, client, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+// request format (comma-separated flags): /auth/featureflags?flags=noobaa,rgw
+func handleGet(w http.ResponseWriter, r *http.Request, client ctrlclient.Client, namespace string) {
+	flagsParam := r.URL.Query().Get("flags")
+	if flagsParam == "" {
+		klog.Errorf("flags parameter is required")
+		http.Error(w, "flags parameter is required", http.StatusBadRequest)
+		return
+	}
+
+	flagNames := strings.Split(flagsParam, ",")
+	for i, flag := range flagNames {
+		flagNames[i] = strings.TrimSpace(flag)
+	}
+
+	type flagResult struct {
+		Value bool   `json:"value"`
+		Error string `json:"error,omitempty"`
+	}
+
+	result := make(map[string]flagResult)
+	for _, flagName := range flagNames {
+		switch flagName {
+		case flagNoobaa:
+			value, err := checkNoobaaFlag(client, namespace, r.Context())
+			if err != nil {
+				klog.Errorf("failed to check noobaa flag: %v", err)
+				result[flagName] = flagResult{Value: false, Error: err.Error()}
+			} else {
+				result[flagName] = flagResult{Value: value}
+			}
+		case flagRGW:
+			value, err := checkRGWFlag(client, r.Context())
+			if err != nil {
+				klog.Errorf("failed to check rgw flag: %v", err)
+				result[flagName] = flagResult{Value: false, Error: err.Error()}
+			} else {
+				result[flagName] = flagResult{Value: value}
+			}
+		default:
+			result[flagName] = flagResult{Value: false, Error: "unsupported flag"}
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	allFlagsFailed := true
+	for _, res := range result {
+		if res.Error == "" {
+			allFlagsFailed = false
+			break
+		}
+	}
+
+	// if all flags failed, returning "500" as response
+	// if some flags succeeded, returning "200" with partial results and error details
+	if allFlagsFailed && len(result) > 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+	} else {
+		w.WriteHeader(http.StatusOK)
+	}
+
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		klog.Errorf("failed to encode response: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+}
+
```

Comment:
too much chaining of functions increasing the call depth. below is my suggestion:
```
func isNoobaaAvailable() {
// do a get on NoobaaCR, it's not about hardcoding it's about not using the info we already know
// do a get on StorageClass with MatchingField on Provisioner (yes, we need cache + index), again we know the name of Provisioner
}

func isRGWAvailable() {
// ditto as above
}
```
ux-backend is an extension of ocs-op and ocs-op is the one creating or dictating the values for the fields that we are interested in and not some other component, not using them is a missed opportunity.

Reply:
https://github.com/red-hat-storage/ocs-operator/pull/3536/files#r2476652372